### PR TITLE
Update NDN stack to 0.8.0 and 22.12 releases

### PR DESCRIPTION
**Pull request by bertof**
###### Description of changes

Updated derivations to latest release, bump to boost version to latest version.

ndn-cxx: 0.7.1 -> 0.8.0
nfd: 0.7.1 -> 22.02
ndn-tools: 0.7.1 -> 22.02

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Software has been tested with default configuration on x86_64-linux

### CODE REVIEW

**Comment by bertof** [comment]
Updates:
- ndn-cxx: 0.7.1 -> 0.8.1
- ndn-tools: 0.7.0 -> 22.12
- nfd: 0.7.1 -> 22.12

tested on NixOS x86_64-linux.

File: pkgs/development/libraries/ndn-cxx/default.nix [hunk 1]
```diff
@@ -6,38 +6,37 @@
...
+    "--with-tests"
   ];
 
-
-  doCheck = false; # disabled since upstream tests fail (Net/TestFaceUri/ParseDev Bug #3896)
+  doCheck = false;
   checkPhase = ''
     runHook preCheck
-    LD_PRELOAD=build/ndn-cxx.so build/unit-tests
+    LD_PRELOAD=build/libndn-cxx.so build/unit-tests # some tests fail in upstream, some fail because of the sandbox environment
```
Comment (NickCao) [inline pkgs/development/libraries/ndn-cxx/default.nix:39]: Move the note to the side of `doCheck = false;`, to make the intention clear.

File: pkgs/tools/networking/ndn-tools/default.nix [hunk 2]
```diff
@@ -12,28 +12,28 @@
...
+    "--boost-includes=${boost.dev}/include"
+    "--boost-libs=${boost.out}/lib"
+    "--with-tests"
   ];
 
   doCheck = false;
   checkPhase = ''
     runHook preCheck
-    build/unit-tests
+    build/unit-tests # some tests fail because of the sandbox environment
```
Comment (NickCao) [inline pkgs/tools/networking/ndn-tools/default.nix:36]: Same here.

**Review by NickCao — COMMENTED** [review]
Apart from the location of the notes, LGTM.

**Comment by bertof** [comment]
Suggestions applied. Thank you.

**Review by NickCao — APPROVED** [review]

**Pull request merged**